src/HashingLogic.ts

```diff
@@ -93,23 +107,64 @@ export interface IAttestation {
   aux: string
 }
 
-export interface IAttestationNode extends IAttestation {
-  link: IRevocationLinks
+export interface IIssuedClaimNode extends IClaimNode {
+  issuance: IIssuanceNode
 }
 
-export interface IDataNode {
-  attestationNode: IAttestationNode
-  signedAttestation: string // Root hash of Attestation tree signed by attester
+export interface ISignedClaimNode {
+  claimNode: IIssuedClaimNode
+  attesterSig: string // Root hash of claim tree signed by attester
+}
+
+export interface IBloomBatchMerkleTreeComponents {
+  batchLayer2Hash: string // Hash of attester sig and subject sig
```

Comment:
Are we still going to do on-chain attestations?

Reply:
This data structure should be agnostic to that. It's usable without putting it on chain but someone could put it on chain if their use case requires it